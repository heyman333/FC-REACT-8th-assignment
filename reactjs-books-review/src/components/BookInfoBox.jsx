import React from "react";
import styled from "styled-components";
import { useDispatch } from "react-redux";
import { Button } from "antd";
import { push } from "connected-react-router";

import { deleteBook } from "../store/book";

const Wrap = styled.button`
  box-sizing: border-box;
  padding: 1.5rem;
  border: 1px black solid;
  display: flex;
  flex-direction: column;
  width: 335px;
  margin: 20px 0px;
  :hover {
    box-shadow: 0px 0px 15px rgb(144, 144, 144);
  }
  :focus {
    outline: 0;
  }
`;

const Info = styled.div`
  display: flex;
  flex-direction: column;
`;

const TitleText = styled.div`
  font-size: 14px;
  font-weight: bold;
`;

const DescText = styled.div`
  font-size: 15px;
  font-weight: bolder;
  color: #ff5244;
`;

const DelButton = styled(Button)`
  width: 55px;
  height: 30px;
`;

const ButtonWrap = styled.div`
  display: flex;
  width: 120px;
  justify-content: space-between;
  margin-top: 20px;
`;

const BookInfoBox = ({ item }) => {
  const dispatch = useDispatch();

  const onDelBook = () => {
    dispatch(deleteBook(item.bookId));
  };

  const onEdit = () => {
    console.log("수정");
  };

  const onClickDetail = () => {
    dispatch(push(`/bookDetail/${item.bookId}`));
  };

  return (
    <Wrap className="bookInfobox" onClick={onClickDetail}>
      <Info>
        <TitleText>책 제목</TitleText>
        <DescText>{item.title}</DescText>

        <TitleText>책 내용</TitleText>
        <DescText>{item.message}</DescText>

        <TitleText>저자</TitleText>
        <DescText>{item.author}</DescText>

        <TitleText>책 링크</TitleText>
        <DescText>{item.url}</DescText>
      </Info>
      <ButtonWrap>
        <DelButton type="danger" onClick={onDelBook}>
          삭제
        </DelButton>
        <DelButton type="default" onClick={onEdit}>
          수정
        </DelButton>
      </ButtonWrap>
    </Wrap>
  );
};

export default BookInfoBox;
