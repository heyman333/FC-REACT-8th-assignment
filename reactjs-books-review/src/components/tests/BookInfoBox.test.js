import React from "react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import BookInfoBox from "../BookInfoBox";

const bookMock = {
  title: "book title",
  message: "book message",
  author: "book author",
  url: "book url",
};

function renderWithRedux(
  ui,
  { initialState, store = createStore(() => {}) } = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}

describe("<BookInfoBox />", () => {
  it("matches snapshot", () => {
    const utils = renderWithRedux(<BookInfoBox item={bookMock} />);
    expect(utils.container).toMatchSnapshot();
  });

  it("shows the book info titles properly", () => {
    const utils = renderWithRedux(<BookInfoBox item={bookMock} />);
    utils.getByText("책 제목");
    utils.getByText("책 내용");
    utils.getByText("저자");
    utils.getByText("책 링크");
  });

  it("shows the delete and edit button", () => {
    const utils = renderWithRedux(<BookInfoBox item={bookMock} />);
    utils.getByText("삭제");
    utils.getByText("수정");
  });
});
