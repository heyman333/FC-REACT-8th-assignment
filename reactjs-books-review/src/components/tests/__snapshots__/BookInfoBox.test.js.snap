// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookInfoBox /> matches snapshot 1`] = `
<div>
  <button
    class="bookInfobox sc-bdVaJa dgxKZg"
  >
    <div
      class="sc-bwzfXH kulSxP"
    >
      <div
        class="sc-htpNat eCmQCD"
      >
        책 제목
      </div>
      <div
        class="sc-bxivhb ipTDf"
      >
        book title
      </div>
      <div
        class="sc-htpNat eCmQCD"
      >
        책 내용
      </div>
      <div
        class="sc-bxivhb ipTDf"
      >
        book message
      </div>
      <div
        class="sc-htpNat eCmQCD"
      >
        저자
      </div>
      <div
        class="sc-bxivhb ipTDf"
      >
        book author
      </div>
      <div
        class="sc-htpNat eCmQCD"
      >
        책 링크
      </div>
      <div
        class="sc-bxivhb ipTDf"
      >
        book url
      </div>
    </div>
    <div
      class="sc-EHOje eFrims"
    >
      <button
        class="ant-btn sc-ifAKCX jGaOAa ant-btn-danger"
        type="button"
      >
        <span>
          삭제
        </span>
      </button>
      <button
        class="ant-btn sc-ifAKCX jGaOAa ant-btn-default"
        type="button"
      >
        <span>
          수정
        </span>
      </button>
    </div>
  </button>
</div>
`;
